### PR TITLE
feat(linter): implement no-labels rule

### PR DESCRIPTION
**Pull request by AnsonH**
## Description

Implements [eslint/no-labels](https://eslint.org/docs/latest/rules/no-labels) rule.

Part of #479

This is my first time contributing to the oxc codebase, so appreciate any comments 🙇

### CODE REVIEW

**Review by camc314 — APPROVED** [review]
thank you!

**Comment by AnsonH** [comment]
(Noob question) I saw that the CI status is stuck at "waiting for approval". Wanna ask do I need to anything special? 

image

**Comment by camc314** [comment]
Running now

**Pull request merged**
